src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/X509Pal.cs

```diff
@@ -38,7 +38,25 @@ public ECDiffieHellman DecodeECDiffieHellmanPublicKey(ICertificatePal? certifica
             public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte[] encodedParameters,
                 ICertificatePal? certificatePal)
             {
-                throw new NotImplementedException(nameof(DecodePublicKey));
+                switch (oid.Value)
+                {
+                    case Oids.Dsa:
+                        throw new NotImplementedException($"{nameof(DecodePublicKey)} (DSA)");
+                    case Oids.Rsa:
+                        if (certificatePal != null)
+                        {
+                            Console.WriteLine($"{nameof(DecodePublicKey)} [with cert pal]");
```

Comment:
I assume we don't want to include the Console.WriteLine calls in the product?

Reply:
Heh, oops.